### DISCUSSION TITLE
Purpose of exception on some methods?

### DISCUSSION THREAD

**Discussion by Gounlaf** [dlemstra/Magick.Native#38] [Q&A] [answered]
:wave: 

I'm not used to C/C++, so my question may be dumb

In the below snippet, I don't understand what the exception is used for:

```c
MAGICK_NATIVE_EXPORT ssize_t MagickImage_ColormapSize_Get(const Image *instance, ExceptionInfo **exception)
{
  (void) exception;

  if (instance->colormap == (PixelInfo *) NULL)
    return -1;

  return (ssize_t) instance->colors;
}
```

What I understand is that it's not used, but I'm not sure.
In that case, could the API be simplified?

Regards.

**Comment by dlemstra** [comment, marked as answer]
This has to do with the fact that the `MagickImage_ColorSpace_Set` requires `exception`. The code generation can only set that exception for both the `get` and `set`. I will probably change the code generator in the future and maybe I can make this possible then but for now you need to pass it.
